src/battle_util.c

```diff
@@ -11085,3 +11085,12 @@ bool32 IsAlly(u32 battlerAtk, u32 battlerDef)
 {
     return (GetBattlerSide(battlerAtk) == GetBattlerSide(battlerDef));
 }
+
+bool32 IsGen6ExpShareEnabled(void)
+{
+#if I_EXP_SHARE_ITEM < GEN_6
+    return FALSE;
+#else
+    return FlagGet(I_EXP_SHARE_FLAG);
+#endif
+}
```

Comment:
I still don't get why this function is returning the opposite value.

Reply:
I don't quite understand what you mean? Seems to be working as I expect but maybe I'm missing something

Reply:
I mean that when I was handling this PR, `IsGen6ExpShareEnabled` was giving me inverted results, which is why you swapped the text/SE/toggle in `ItemUseOutOfBattle_ExpShare`.

Reply:
Oh interesting. No idea why, but glad it's gone now? 🤷

Reply:
No, but the issue is that `!IsGen6ExpShareEnabled()` in `ItemUseOutOfBattle_ExpShare` turns the Exp Share *Off* for some reason

Reply:
`FlagToggle` is what turns it off, _before_ `!IsGen6ExpShareEnabled()`

Lines 211 - 215 are just informing the player,
> hey this is different now

Reply:
*Ooooooooooh* that's why ^^;
I think the Lunos branch had it so the flag set and clears were inside the if/else, and I changed it to a toggle for some reason.

To make it clearer when reading the code, I would either put the toggle below the ifdef and remove the `!` from the `if`.

Reply:
To be clear, not in this function, but in `ItemUseOutOfBattle_ExpShare` instead.